### DISCUSSION TITLE
怎么计算文本内容的宽度?

### DISCUSSION THREAD

**Discussion by keman5** [antvis/G2#5314] [Q&A]
G2内部有没有API可以计算文本宽高? 有很多时候需要1行显示不同样式的文本, 那就需要计算前面文本的宽度, 文档中没有找到相关API

**Comment by pearmini** [comment]
能提供更加具体的场景吗？

**Reply by keman5** [reply]
例如饼图中心需要显示带单位, 但是单位需要比这行文本字体小很多, 直接字符串拼接起来就不能使用两种字体大小, 所以需要两个 annotation().text, 那么单位的position就要根据前面的内容进行调整

**Reply by pearmini** [reply]
可以用 annotation.html 添加富文本？

**Reply by keman5** [reply]
文档里没有相关内容, 有这种案例吗?
